sdk/appcenter/src/androidTest/java/com/microsoft/appcenter/utils/storage/DatabaseManagerAndroidTest.java

```diff
@@ -503,6 +506,89 @@ public void testMultipleTablesCreateDelete() {
         assertFalse(checkTableExists(databaseManager, secondTable));
     }
 
+    @Test
+    public void replaceInTableWithUniquenessConstraint() {
+        String tableName = "myTable";
+        ContentValues content1 = new ContentValues();
+        ContentValues content2 = new ContentValues();
+        String colStr = "colStr";
+        content1.put(colStr, "first string");
+        content2.put(colStr, "different string");
+        String colInt = "colInt";
+        content1.put(colInt, -1);
+        content2.put(colInt, -1);
+        String colBoolean = "colBoolean";
+        content1.put(colBoolean, true);
+        content2.put(colBoolean, true);
+        DatabaseManager databaseManager = insertContentValuesAndVerifyContentValuesInserted("replace-with-uniqueness-constraint", tableName, content1, colInt, colBoolean, colStr);
+
+        /* Should replace first row because they have the same values in the unique columns. */
+        long result = databaseManager.replace(tableName, content2);
+        assertTrue(result > 0);
+        assertEquals(1L, databaseManager.getRowCount());
+        Cursor cursor = databaseManager.getCursor(null, null, null, null);
+        ContentValues row = databaseManager.nextValues(cursor);
+        assertNotNull(row);
+        assertEquals("different string", row.getAsString(colStr));
+        assertEquals(-1, row.getAsInteger(colInt).intValue());
+        assertEquals(true, row.getAsBoolean(colBoolean));
+    }
+
+    private DatabaseManager insertContentValuesAndVerifyContentValuesInserted(String databaseName, String tableName, ContentValues content1, String colInt, String colBoolean, String colStr) {
```

Comment:
Could we move the helper method to the end of the test file?

Reply:
IMO it's more of a developer's preference of where to put helper methods. Placing it at the end of file makes it easier to skim through all the tests, while keeping it close to the caller makes it easier to view in editor. I don't have a strong preference either way.